hal/armv7a/imx6ull/_init.S

```diff
@@ -353,6 +355,34 @@ _cpy4:
 _start:
 	cpsid aif, #SYS_MODE
 
+#ifdef PLO_BOOT
+	ldr r0, =relOffs
+	ldr r1, =#VADDR_SYSPAGE
+	lsr r1, #12
+	lsl r1, #12
+
+	sub r0, r0, #VADDR_KERNEL
+	add r0, r0, #ADDR_DDR
+	sub r2, r1, r9
+	str r2, [r0]
+
+	ldr r0, [r9, #4] /* syspage size address - syspage address + sizeof(hal_syspage_t) */
+	add r2, r9, r0   /* end of the syspage */
+
+	ldr r0, =syspage
+	sub r0, r0, #VADDR_KERNEL
+	add r0, r0, #ADDR_DDR
+	str r1, [r0]
+
+	sub r1, r1, #VADDR_KERNEL
+	add r1, r1, #ADDR_DDR
```

Comment:
Looks weird, but should work. Because of weirdness just a suggestion
```suggestion
	add r1, r1, #(ADDR_DDR - VADDR_KERNEL)
```
`ldr r1, =(VADDR_SYSPAGE - VADDR_KERNEL + ADDR_DDR)` is more code, but also valid and somewhat more readable